apis/outlook/calendar/code/pkg/graph/events.go

```diff
@@ -0,0 +1,287 @@
+package graph
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gptscript-ai/tools/apis/outlook/calendar/code/pkg/util"
+	msgraphsdkgo "github.com/microsoftgraph/msgraph-sdk-go"
+	"github.com/microsoftgraph/msgraph-sdk-go/groups"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/microsoftgraph/msgraph-sdk-go/users"
+)
+
+type CreateEventInfo struct {
+	Attendees                   []string // slice of email addresses
+	Subject, Location, Body, ID string
+	Owner                       OwnerType
+	IsOnline                    bool
+	Start, End                  time.Time
+}
+
+func GetEvent(ctx context.Context, client *msgraphsdkgo.GraphServiceClient, eventID, calendarID string, owner OwnerType) (models.Eventable, error) {
+	if calendarID != "" {
+		switch owner {
+		case OwnerTypeUser:
+			resp, err := client.Me().Calendars().ByCalendarId(calendarID).Events().ByEventId(eventID).Get(ctx, &users.ItemCalendarsItemEventsEventItemRequestBuilderGetRequestConfiguration{})
+			if err != nil {
+				return nil, fmt.Errorf("failed to get event: %w", err)
+			}
+			return resp, nil
+		case OwnerTypeGroup:
+			resp, err := client.Groups().ByGroupId(calendarID).Events().ByEventId(eventID).Get(ctx, &groups.ItemEventsEventItemRequestBuilderGetRequestConfiguration{})
+			if err != nil {
+				return nil, fmt.Errorf("failed to get event: %w", err)
+			}
+			return resp, nil
+		}
+	}
+
+	resp, err := client.Me().Events().ByEventId(eventID).Get(ctx, &users.ItemEventsEventItemRequestBuilderGetRequestConfiguration{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get event: %w", err)
+	}
+
+	return resp, nil
+}
+
+func CreateEvent(ctx context.Context, client *msgraphsdkgo.GraphServiceClient, info CreateEventInfo) (models.Eventable, error) {
+	requestBody := models.NewEvent()
+
+	var attendees []models.Attendeeable
+	for i := range info.Attendees {
+		a := info.Attendees[i]
```

Comment:
As of Go 1.22, you don't have do this.

Reply:
Thanks. I heard that this got changed at some point but didn't know for sure if it was safe yet.

Reply:
Fixed